src/convertToObject.js

```diff
@@ -6,7 +6,44 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString === '') {
+    return {};
+  }
+
+  const modifiedStr = sourceString
+    .trim()
+    .split(';')
+    .map((item) => {
+      return item.trim();
+    });
+
+  if (modifiedStr.length === 1) {
+    const [key, value] = sourceString.trim().split(':');
```

Comment:
The condition `if (modifiedStr.length === 1)` is misleading. It should check if the string contains only one key-value pair by checking if `sourceString` contains a single colon. Consider using `sourceString.includes(':')` instead.